index.js

```diff
@@ -377,7 +377,68 @@ instance.prototype.actions = function(system) {
 				}
 			]
 		},
-		'stageDisplayHideMessage': { label: 'Stage Display Hide Message' }
+		'stageDisplayHideMessage': { label: 'Stage Display Hide Message' },
+		'clockStart': {
+			label: 'Start Clock',
+			options: [
+				{
+					type: 'textinput',
+					label: 'Clock Number',
+					id: 'clockIndex',
+					default: 0,
+					regex: self.REGEX_SIGNED_NUMBER
+				}
+			]
+		},
+		'clockStop': {
+			label: 'Stop Clock',
+			options: [
+				{
+					type: 'textinput',
+					label: 'Clock Number',
+					id: 'clockIndex',
+					default: 0,
+					regex: self.REGEX_SIGNED_NUMBER
+				}
+			]
+		},
+		'clockReset': {
+			label: 'Reset Clock',
+			options: [
+				{
+					type: 'textinput',
+					label: 'Clock Number',
+					id: 'clockIndex',
+					default: 0,
+					regex: self.REGEX_SIGNED_NUMBER
+				}
+			]
+		},
+		'clockUpdate': {
+			label: 'Update CountDown Clock',
+			options: [
+				{
+					type: 'textinput',
+					label: 'Clock Number',
+					id: 'clockIndex',
+					default: 0,
+					regex: self.REGEX_SIGNED_NUMBER
+				},
+				{
+					type: 'textinput',
+					label: 'Clock Time',
+					id: 'clockTime',
+					default: "00:05:00",
```

Comment:
Could you add a regex check on the clock time option, something like:
`regex: '/^\d{2}:\d{2}:\d{2}$/'`

It wouldn't validate the correctness of the time, but it would stop the command from getting corrupted if strange characters got in there.

Reply:
I might go with this....
regex: '/^\\d*:?\\d*:?\\d*$/'

Because Pro6 seems happy to calculate the correct time even with values greater than 60 units in either of the three placeholders.

For example:
00:05:00 | 05:00 | 05: | 300 = 5 Minutes
10:00:00 | 10:00: | 10:: | 600:00 | 600:: | 36000 = 10hrs

It seems to handled as follows...
Any number with no ":" following it is calculated as seconds.
Any number in front of one ":" is calculated as minutes (even for values greater than 60)
Any number in front of two ":" is calculated as hours (even for values greater than 60)

It seems quite flexible with the calculation and allowing shorthand - I don't mind it myself.
What do you think? 
* Keep ultimate flexibility? regex: '/^\\d*:?\\d*:?\\d*$/'
* Allow shorthand (but limit to 2 digits)? regex: '/^\\d{0,2}:?\\d{0,2}:?\\d{0,2}$/'
* Should we arbitrarily enforce HH:MM:SS? regex: '/^\\d{2}:\\d{2}:\\d{2}$/'

Reply:
If Pro6 is flexible with calculating time from the input, then we should extend that flexibility to the user. My only concern was the user entering something like `"}` and corrupting the JSON. (I have seen invalid input outright crashing Pro6 before).

So let's go with the first option like you have,

Great work on the PR 👍

Reply:
Validation is now added to the duration input. I have also added some minor updates (tooltips,  and updated a label) and also updated Help.  I have completed some informal testing and so far it works well.